sycl/include/sycl/ext/intel/fpga_device_selector.hpp

```diff
@@ -22,6 +22,29 @@ class platform;
 namespace ext {
 namespace intel {
 
+namespace detail {
+// Scores a device by platform name. Uses const char * for the platform name to
+// avoid repeated string copying.
+inline int SelectDeviceByPlatform(const char *required_platform_name,
+                                  const device &device) {
+  const platform &pf = device.get_platform();
+  const std::string &platform_name = pf.get_info<sycl::info::platform::name>();
```

Comment:
sycl::info::platform::name query returns 'std::string'.
Why do you need '&' before 'platform_name' ?